collector/database/chart.php

```diff
@@ -107,25 +106,69 @@
 
 $periodselect->set_label(get_string('select_graph_period', 'cltr_database'));
 
+$configfrequency = get_config('tool_cloudmetrics', $metricname . '_frequency');
```

Comment:
You should not use get_config() to obtain or set values associated with metrics. Not all metrics will be defined in the tool_cloudmetrics plugin, so this line line would not work for non-native metrics. Use the metric classes instead.

```suggestion
$configfrequency = $metrics[$metricname]->get_frequency();
```